src/helpers/operations.ts

```diff
@@ -156,20 +156,45 @@ export const handleSequence = (input: Array<any>, fns: ((input: Array<any>) => A
  * @throws Error when attempting to map over non-array and non-object values
  */
 export const handleMap = (input: Array<any>, fn: (input: any) => any): Array<any> => {
+// Results array will hold one array containing all mapped values
   const results = []
+  const mappedValues = []
 
   for (const item of input) {
     if (Array.isArray(item)) {
-      const result = fn(item)
-      results.push(result)
-    } else if (typeof item === 'object') {
-      const result = fn(Object.values(item))
-      results.push(result)
+      // For arrays, apply the function directly to each element
+      for (const element of item) {
+        // Get all results from applying the function to this element
+        const fnResults = fn([element])
```

Comment:
it seems a bit convoluted, maybe I miss the context

Reply:
It is. I've added some context and explanation.